### PR TITLE
fix(@wordpress/components): make `Id` prop optional in the `BaseControl` component

### PR DESCRIPTION
**Pull request by rafaucau**
The `Id` attribute used to be required, but is now optional: 

- https://github.com/WordPress/gutenberg/blob/trunk/packages/components/src/base-control/README.md#id
- https://github.com/WordPress/gutenberg/blob/4056660341852e2b627127d2dacbbedd38cb6e3a/packages/components/src/base-control/types.ts#L20

### CODE REVIEW

**Comment by rafaucau** [comment]
I just noticed that [@wordpress/components](https://www.npmjs.com/package/@wordpress/components) already has types. Perhaps it would be better to remove it from DefinitelyTyped?
https://unpkg.com/browse/@wordpress/components@27.2.0/build-types/

File: types/wordpress__components/base-control/index.d.ts [hunk 1]
```diff
@@ -27,10 +27,11 @@ declare namespace BaseControl {
...
     interface Props extends ControlProps {
         /**
-         * The id of the element to which labels and help text are being
-         * generated. That element should be passed as a child.
+         * The HTML `id` of the control element (passed in as a child to `BaseControl`) to which labels and help text are being generated. This is necessary to accessibly associate the label with that element.
+         *
+         * The recommended way is to use the `useBaseControlProps` hook, which takes care of generating a unique `id` for you. Otherwise, if you choose to pass an explicit `id` to this prop, you are responsible for ensuring the uniqueness of the `id`.
          */
-        id: string;
+        id?: string | undefined;
```
Comment (manzoorwanijk) [inline types/wordpress__components/base-control/index.d.ts:34]: Optional prop already implies `undefined`

```suggestion
        id?: string;
```

**Review by manzoorwanijk — APPROVED** [review]
This looks good to me. Thank you.

**Comment by manzoorwanijk** [comment]
> I just noticed that [@wordpress/components](https://www.npmjs.com/package/@wordpress/components) already has types. Perhaps it would be better to remove it from DefinitelyTyped? https://unpkg.com/browse/@wordpress/components@27.2.0/build-types/

Yes, it can be marked as deprecated.

**Comment by rafaucau** [comment]
Ready to merge

**Comment by rafaucau** [comment]
> @manzoorwanijk  Yes, it can be marked as deprecated.

Ok, I have started to remove the types here: #69133

**Pull request merged**